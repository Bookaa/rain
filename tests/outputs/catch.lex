keyword 'import'
name 'string'
newline
name 'three'
symbol '='
keyword 'func'
symbol '('
symbol ')'
indent
name 'panic'
symbol '('
string 'three!'
symbol ')'
newline
dedent
newline
name 'two'
symbol '='
keyword 'func'
symbol '('
symbol ')'
indent
keyword 'let'
name 'x'
symbol '='
name 'three'
symbol '?'
symbol '('
symbol ')'
newline
name 'print'
symbol '('
string 'two caught '
operator '$'
name 'string'
symbol '.'
name 'tostr'
symbol '('
name 'x'
symbol ')'
symbol ')'
newline
dedent
newline
name 'one'
symbol '='
keyword 'func'
symbol '('
symbol ')'
indent
keyword 'catch'
name 'x'
indent
name 'two'
symbol '('
symbol ')'
newline
dedent
newline
name 'print'
symbol '('
string 'one caught '
operator '$'
name 'string'
symbol '.'
name 'tostr'
symbol '('
name 'x'
symbol ')'
symbol ')'
newline
dedent
newline
name 'main'
symbol '='
keyword 'func'
symbol '('
symbol ')'
indent
name 'print'
symbol '('
string '----'
symbol ')'
newline
keyword 'catch'
name 'x'
indent
name 'print'
symbol '('
string 'before'
symbol ')'
newline
name 'one'
symbol '('
symbol ')'
newline
name 'print'
symbol '('
string 'after'
symbol ')'
newline
dedent
newline
name 'print'
symbol '('
string 'main caught '
operator '$'
name 'string'
symbol '.'
name 'tostr'
symbol '('
name 'x'
symbol ')'
symbol ')'
newline
name 'print'
symbol '('
string '----'
symbol ')'
newline
name 'print'
symbol '('
string 'before'
symbol ')'
newline
name 'x'
symbol '='
name 'one'
symbol '?'
symbol '('
symbol ')'
newline
name 'print'
symbol '('
string 'after'
symbol ')'
newline
name 'print'
symbol '('
string 'main caught '
operator '$'
name 'string'
symbol '.'
name 'tostr'
symbol '('
name 'x'
symbol ')'
symbol ')'
newline
name 'print'
symbol '('
string '----'
symbol ')'
newline
keyword 'catch'
name 'x'
indent
name 'print'
symbol '('
string 'before'
symbol ')'
newline
name 'panic'
symbol '('
string 'obvious'
symbol ')'
newline
name 'print'
symbol '('
string 'after'
symbol ')'
newline
dedent
newline
name 'print'
symbol '('
string 'main caught '
operator '$'
name 'string'
symbol '.'
name 'tostr'
symbol '('
name 'x'
symbol ')'
symbol ')'
newline
name 'print'
symbol '('
string '----'
symbol ')'
newline
name 'print'
symbol '('
string 'before'
symbol ')'
newline
name 'x'
symbol '='
name 'panic'
symbol '?'
symbol '('
string 'obvious'
symbol ')'
newline
name 'print'
symbol '('
string 'after'
symbol ')'
newline
name 'print'
symbol '('
string 'main caught '
operator '$'
name 'string'
symbol '.'
name 'tostr'
symbol '('
name 'x'
symbol ')'
symbol ')'
newline
dedent
newline
EOF
