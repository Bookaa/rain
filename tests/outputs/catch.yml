!program;1
stmts:
- !assn;2
  export: false
  let: true
  lhs: !name;1
    value: three
  rhs: !func;1
    body: !block;1
      stmts:
      - !call;1
        args:
        - !str;1
          value: three!
        catch: false
        func: !name;1
          value: panic
    params: []
- !assn;2
  export: false
  let: true
  lhs: !name;1
    value: two
  rhs: !func;1
    body: !block;1
      stmts:
      - !assn;2
        export: false
        let: true
        lhs: !name;1
          value: x
        rhs: !call;1
          args: []
          catch: true
          func: !name;1
            value: three
      - !call;1
        args:
        - !binary;1
          lhs: !str;1
            value: 'two caught '
          op: $
          rhs: !call;1
            args:
            - !name;1
              value: x
            catch: false
            func: !name;1
              value: tostr
        catch: false
        func: !name;1
          value: print
    params: []
- !assn;2
  export: false
  let: true
  lhs: !name;1
    value: one
  rhs: !func;1
    body: !block;1
      stmts:
      - !catch;1
        body: !block;1
          stmts:
          - !call;1
            args: []
            catch: false
            func: !name;1
              value: two
        name: x
      - !call;1
        args:
        - !binary;1
          lhs: !str;1
            value: 'one caught '
          op: $
          rhs: !call;1
            args:
            - !name;1
              value: x
            catch: false
            func: !name;1
              value: tostr
        catch: false
        func: !name;1
          value: print
    params: []
- !assn;2
  export: false
  let: true
  lhs: !name;1
    value: main
  rhs: !func;1
    body: !block;1
      stmts:
      - !call;1
        args:
        - !str;1
          value: '----'
        catch: false
        func: !name;1
          value: print
      - !catch;1
        body: !block;1
          stmts:
          - !call;1
            args:
            - !str;1
              value: before
            catch: false
            func: !name;1
              value: print
          - !call;1
            args: []
            catch: false
            func: !name;1
              value: one
          - !call;1
            args:
            - !str;1
              value: after
            catch: false
            func: !name;1
              value: print
        name: x
      - !call;1
        args:
        - !binary;1
          lhs: !str;1
            value: 'main caught '
          op: $
          rhs: !call;1
            args:
            - !name;1
              value: x
            catch: false
            func: !name;1
              value: tostr
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !str;1
          value: '----'
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !str;1
          value: before
        catch: false
        func: !name;1
          value: print
      - !assn;2
        export: false
        let: false
        lhs: !name;1
          value: x
        rhs: !call;1
          args: []
          catch: true
          func: !name;1
            value: one
      - !call;1
        args:
        - !str;1
          value: after
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !binary;1
          lhs: !str;1
            value: 'main caught '
          op: $
          rhs: !call;1
            args:
            - !name;1
              value: x
            catch: false
            func: !name;1
              value: tostr
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !str;1
          value: '----'
        catch: false
        func: !name;1
          value: print
      - !catch;1
        body: !block;1
          stmts:
          - !call;1
            args:
            - !str;1
              value: before
            catch: false
            func: !name;1
              value: print
          - !call;1
            args:
            - !str;1
              value: obvious
            catch: false
            func: !name;1
              value: panic
          - !call;1
            args:
            - !str;1
              value: after
            catch: false
            func: !name;1
              value: print
        name: x
      - !call;1
        args:
        - !binary;1
          lhs: !str;1
            value: 'main caught '
          op: $
          rhs: !call;1
            args:
            - !name;1
              value: x
            catch: false
            func: !name;1
              value: tostr
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !str;1
          value: '----'
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !str;1
          value: before
        catch: false
        func: !name;1
          value: print
      - !assn;2
        export: false
        let: false
        lhs: !name;1
          value: x
        rhs: !call;1
          args:
          - !str;1
            value: obvious
          catch: true
          func: !name;1
            value: panic
      - !call;1
        args:
        - !str;1
          value: after
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !binary;1
          lhs: !str;1
            value: 'main caught '
          op: $
          rhs: !call;1
            args:
            - !name;1
              value: x
            catch: false
            func: !name;1
              value: tostr
        catch: false
        func: !name;1
          value: print
    params: []
