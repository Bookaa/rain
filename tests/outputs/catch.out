----
before
two caught three!
one caught null
after
main caught null
----
before
two caught three!
one caught null
after
main caught null
----
before
main caught obvious
----
before
after
main caught obvious
